wordpress-mobile/WordPress-Android
# Select tab on button tap

Fixes #11849 

This PR is part 5.5 of a larger PR that moves the tab layout to a parent.

The approach here was to pass the `ReaderViewModel` to the `ReaderPostListViewModel` and when an empty view button is clicked, send the request back to the observer (`ReaderFragment`) who will then change the selected index of view_pager. This way the normal viewpager handler will take over as if the tab itself was tapped. Note that we do save change the selectedTag value before posting the button tap to the viewmodel.

This may or may not be the best approach, but without knowing what else will change, I felt it was the safest option at the moment. We can revisit the further we get with the tab layout transformation.

Merge Instructions
1. Review this PR
2. Make sure #11890 is merged
3. Change the target branch to master-feature-branch/move-reader-tablayout-to-parent
4. Remove the "Not Ready for merge" label
5. Merge this PR

To test:
1. Log in as a user who is not following any sites
2. Select "Following" tab in the Reader
3. Tap on the Discover button from within the empty slate
4. Notice the "Discover" tab is selected

1. Log in as a user who has not saved any sites
2. Select "Saved" tab in the Reader
3. Tap on the Following button from within the empty slate
4. Notice the "Following" tab is selected
PR submission checklist:
